### ISSUE TITLE
Confusing Excess Speech Output when attempting to Navigate in a way that isn't possible.

### ISSUE THREAD

**Issue by jmwmusic** [musescore/MuseScore#30463]
### Issue type

Accessibility issue (e.g. for keyboard-only or screen reader users)

### Description with steps to reproduce

When you try to navigate in a way that is not possible the speech used to say “blank” (see [20732](https://github.com/musescore/MuseScore/issues/20732)).  Since the 4.6 update it now has confusing and excessive speech.  
I’ve opened this as a new issue so it can be reviewed as a regression as it is quite different output to the original issue and this excess speech is quite difficult to work with, the “blank” was not a high priority.

It seems to be stating the function of the key press in each situation.

1.	 Place focus on the first note or rest of the score and press left arrow.
You hear “Previous Chord/Shift Text Left” 
2.	  Place the focus on the last note/rest of the score and press right arrow.
You hear “Next Chord/Shift Text Right” 
3.	 Place focus on the top score on the highest note of a chord/top voice and press Alt up arrow.  You hear “Up Note in Chord”. 
4.	Place focus on the bottom note of bottom staff and Press Alt down arrow  and you hear “Down Note in Chord” 

Ideally this could state something concise like “Start”, “End”, “Top, “Bottom, or as there are other instances could it make a system sound, the quiet bong that Microsoft uses when you can’t do an action would be better for this and all other cases.  All users would hear this as it wouldn’t be screen reader specific, but might be useful to all.  Failing these having no output at all would be better than a load of confusing excess speech.

### Supporting files, videos and screenshots

N/A

### In which versions of MuseScore Studio is this issue present?

4.6.1

### Regression

Yes, this used to work in a previous version of MuseScore 4.x

### Operating system

Windows 11 with NVDA

### Checklist

- [x] This report follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting bugs and issues
- [x] I have verified that this issue has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar issues
- [x] I have attached all requested files and information to this report
- [x] I have attempted to identify the root problem as concisely as possible, and have used minimal reproducible examples where possible

**Comment by jrbowden** [comment]
One possible solution to this is based on checking if "anything happened" when a key is pressed.
"Aynthing happened" basically means either the score was changed or the focus position has changed (or both).
To check if the score has changed, could you check whether the Undo stack has changed?

- If position changed: report the new note and position (as at present)
- If score has changed: report what happened:
     Either this will be speech from the function itself, or if not, then fallback is to announce the key function.
- If nothing changed (no change or position or score): say nothing, or make error sound.

This approach will fix the announcement at start/end of score, top/bottom of chords etc, giving a big improvement.
However, it will mean that the following keys need to announce something rather than relying on default behaviour:
- Duration change keys, including dot, especially in Note Input mode (because no score change or focus change has occurred)
- Toggle keys (such as staccato, accent, brackets, etc. (because announcing the function name doesn't tell me whether the toggle is on or off).

**Comment by LeonarddeR** [comment]
The excessive speech is caused by it being set as the accessible label of the score view. This is strictly spoken completely unrelated to the blank announcements, rather a new regression.
